mqtt3/1.0/modules/ROOT/edits/mqtt3-connector-listener.adoc

```diff
@@ -0,0 +1,156 @@
+. Set *QoS* to any of the following quality of service levels:
++
+* *AT_MOST_ONCE*
+* *AT_LEAST_ONCE*
+* *EXACTLY_ONCE*
+. Click *Finish*.
+
+image::mqtt3-onnewmessage.png[MQTT On New Message source configuration in Studio with Topic filters configured]
+
+
+In the Configuration XML editor, the `<mqtt3:listener>`, `<mqtt3:topic>`, and `topicFilter` configurations look like this:
+
+[source,xml,linenums]
+----
+    <flow name="listenerAuthorQuotes">
+        <mqtt3:listener config-ref="MQTT_Config">
+            <mqtt3:topics>
+                <mqtt3:topic topicFilter="quotes/terryPratchett" qos="EXACTLY_ONCE"/>
+                <mqtt3:topic topicFilter="quotes/neilGaiman" qos="AT_LEAST_ONCE" />
+                <mqtt3:topic topicFilter="quotes/ianMcEwan" qos="AT_MOST_ONCE" />
+            </mqtt3:topics>
+        </mqtt3:listener>
+    </flow>
+----
+
+== Configure Single-Level Wildcards in Topic Filters
+
+Configure single-level wildcards to listen to multiple topics. Single-level wildcards (`+`) enable the subscriber to receive messages published to all topics that match a specific structure.
+
+In the following example, you configure the `topicFilter` parameter with the structure `quotes/+/authors`:
+
+[source,xml,linenums]
+----
+    <flow name="quotes">
+        <mqtt3:listener config-ref="${config}">
+            <mqtt3:topics>
+                <mqtt3:topic topicFilter="quotes/+/authors"/>
+            </mqtt3:topics>
+        </mqtt3:listener>
+
+        <logger level="DEBUG" message="A quote has been published to #[attributes.topic]: #[payload]"/>
+    </flow>
+----
+
+Messages published to the following topics trigger the listener:
+
+* `quotes/british/authors`
+* `quotes/american/authors`
+
+However, messages published to the following topics do not trigger the listener:
+
+* `names/british/authors`
+* `quotes/american/writers`
+
+
+== Configure Multi-Level Wildcards in Topic Filters
+
+Configure multi-level wildcards to enable the listener to subscribe to all topics that share the same root (everything that comes before the hash `#` symbol).
+
+In the following example, you configure the `topicFilter` parameter with the structure `quotes/england/#`:
+
+[source,xml,linenums]
+----
+    <flow name="listenerArgentinaTemperature">
+        <mqtt3:listener config-ref="${config}">
+            <mqtt3:topics>
+                <mqtt3:topic topicFilter="quotes/england/#"/>
+            </mqtt3:topics>
+        </mqtt3:listener>
+
+        <logger level="DEBUG" message="A quote has been published to #[attributes.topic]: #[payload]"/>
+    </flow>
+----
+
+Messages published to the following topics trigger the listener:
+
+* `quotes/england/authors/terryPratchett`
+* `quotes/england/authors/neilGaiman`
+* `quotes/england/actors`
+
+However, messages published to the following topics do not trigger the listener:
+
+* `quotes/american/actors`
+* `phrases/england/authors/neilGaiman`
```

Comment:
I really like this structure of showing what "works" and what doesn't; it really helps the audience to understand much better than if you had just included the former!